### PR TITLE
Test for python3.8

### PR DESCRIPTION
**Pull request by hndgzkn**
Update workflows to use Python 3.8.
Fix problems while testing with Python3.8
Fix linting for Python3.8

Tests run without problems for 3.7 as well.

### CODE REVIEW

File: alphacsc/learn_d_z_multi.py [hunk 1]
```diff
@@ -145,8 +145,8 @@ def learn_d_z_multi(X, n_atoms, n_times_atom, n_iter=60, n_jobs=1,
     """
 
     assert lmbd_max in ['fixed', 'scaled', 'per_atom', 'shared'], (
-        "lmbd_max should be in {'fixed', 'scaled', 'per_atom', 'shared'}, "
-        "not '{}'".format(lmbd_max)
+        "lmbd_max should be in {'fixed', 'scaled', 'per_atom', 'shared'}"
+        f"not '{lmbd_max}'"
```
Comment (agramfort) [inline alphacsc/learn_d_z_multi.py:149]:
```suggestion
        "lmbd_max should be in {'fixed', 'scaled', 'per_atom', 'shared'}, "
        f"got '{lmbd_max}'"
```

File: alphacsc/datasets/hcp.py [hunk 2]
```diff
@@ -221,7 +221,7 @@ def make_array(X, equalize='zeropad'):
                            axis=-1) for x in X
         ])
     else:
-        raise ValueError("The equalize '{}' is not valid. It should be in "
-                         "{'crop', 'zeropad'}".format(equalize))
+        raise ValueError(f"The equalize '{equalize}' is not valid. "
+                         "It should be in {'crop', 'zeropad'}")
```
Comment (agramfort) [inline alphacsc/datasets/hcp.py:225]:
```suggestion
        raise ValueError(f"The equalize parameter should be in "
                         f"{'crop', 'zeropad'}, got '{equalize}'.")
```

**Review by agramfort — APPROVED** [review]

File: alphacsc/datasets/hcp.py [hunk 3]
```diff
@@ -221,7 +221,7 @@ def make_array(X, equalize='zeropad'):
                            axis=-1) for x in X
         ])
     else:
-        raise ValueError(f"The equalize '{equalize}' is not valid. "
-                         "It should be in {'crop', 'zeropad'}")
+        raise ValueError(f"The equalize parameter should be in "
```
Comment (tomMoral) [inline alphacsc/datasets/hcp.py:224]:
```suggestion
        raise ValueError("The equalize parameter should be in "
```

File: .github/workflows/unittests.yml [hunk 4]
```diff
@@ -25,10 +25,10 @@ jobs:
           conda --version
           which python
           pip install numpy cython
-          pip install -e .
           pip install pytest">=3.6" coverage pytest-cov coveralls
       - name: Run unit tests
         run: |
+          pip install -e .
```
Comment (tomMoral) [inline .github/workflows/unittests.yml:31]: Not sure why you change this?
If no particular reason, I would keep the `install` in the `Installation` step.

**Review by tomMoral — APPROVED** [review]
LGTM, just 2 coments

File: .github/workflows/unittests.yml (hunk 4 above)
Comment (hndgzkn) [reply .github/workflows/unittests.yml:31]: I thought of installation step as installation of dependencies. 
If you prefer the previous version, I'll revert that change.

Comment (tomMoral) [reply .github/workflows/unittests.yml:31]: Let's keep like this then :)
Not really a big deal.

**Comment by tomMoral** [comment]
Merging, thx @hndgzkn .

**Pull request merged**